tenets/codelingo/code-review-comments/named-result-params/codelingo.yaml

```diff
@@ -0,0 +1,68 @@
+funcs:
+  - name: isSameName
+    type: asserter
+    body: |
+      function(a, b) {
+        return a.toLowerCase() === b.toLowerCase()
+      }
+tenets:
+  - name: named-result-params
+    flows:
+      codelingo/review:
+        comment: Consider what it will look like in godoc. {{returnName}} {{returnType}} will stutter in godoc; better to use {{returnType}}. From [Code Review Comments](https://github.com/golang/go/wiki/CodeReviewComments#named-result-parameters)
+      codelingo/docs:
+        title: Named Result Parameters
+        body: |
+          Consider what it will look like in godoc. Named result parameters like:
+
+          ```
+          func (n *Node) Parent1() (node *Node)
+          func (n *Node) Parent2() (node *Node, err error)
+          ```
+
+          will stutter in godoc; better to use:
+
+          ```
+          func (n *Node) Parent1() *Node
+          func (n *Node) Parent2() (*Node, error)
+          ```
+
+          On the other hand, if a function returns two or three parameters of the same type, or if the meaning of a result isn't clear from context, adding names may be useful in some contexts. Don't name result parameters just to avoid declaring a var inside the function; that trades off a minor implementation brevity at the cost of unnecessary API verbosity.
+          
+          ```
+          func (f *Foo) Location() (float64, float64, error)
+          ```
+
+          is less clear than:
+
+          ```
+          // Location returns f's latitude and longitude.
+          // Negative values mean south and west, respectively.
+          func (f *Foo) Location() (lat, long float64, err error)
+          ```
+          
+          Naked returns are okay if the function is a handful of lines. Once it's a medium sized function, be explicit with your return values. Corollary: it's not worth it to name result parameters just because it enables you to use naked returns. Clarity of docs is always more important than saving a line or two in your function.
+
+          Finally, in some cases you need to name a result parameter in order to change it in a deferred closure. That is always OK.
+    query: |
+      import codelingo/ast/go
+
+      go.func_decl(depth = any):
+        go.func_type:
+          go.field_list:
+            sibling_order == 1
+            @review comment
+            go.field:
+              child_count == 2
+              go.names:
+                go.ident:
+                  name as returnName
+              any_of:
+                go.star_expr:
+                  go.ident:
+                    name as returnType
+                    isSameName(returnName, returnType)
```

Comment:
This seems like a very specific case. Where is it stipulated in CRC that this only applies where the names are the same?

Reply:
Pretty sure that's what it's implying. The stuttering being the same word twice. See doc in yaml file.

Reply:
OK, I figured they just did that because it was easier. It doesn't say that they need to be similar explicitly, does it?

Also, this won't catch `(err error)`.

Reply:
OK, I figured they just did that because it was easier. It doesn't say that they need to be similar explicitly, does it?

Also, this won't catch `(err error)`.

Reply:
No, it won't catch that but that is intentional as err error is common and often used for returning errors from deferred functions. Thought it would be best to exclude that one.

Reply:
I don't have a strong opinion on whether it should be excluded. But the doc specifically says it would be better to use `error`, and I think we should editorialise as little as possible https://github.com/codelingo/codelingo/pull/375/files#diff-e825bc2f5888babf8dfaf4dae65f36e9R27